### DISCUSSION TITLE
http://lotabout.github.io/2016/write-a-C-interpreter-7/

### DISCUSSION THREAD

**Discussion by lotabout** [lotabout/lotabout.github.io#18] [Announcements]
# http://lotabout.github.io/2016/write-a-C-interpreter-7/
手把手教你构建 C 语言编译器（7）- 语句
http://lotabout.github.io/2016/write-a-C-interpreter-7/

**Comment by lotabout** [comment]
由 Bin Mu 发布于 2019-06-15T02:41:19Z：

..

**Comment by lotabout** [comment]
由 Bin Mu 发布于 2019-06-15T02:42:20Z：

do while 语句：

` else if(token == Do){ ////////////////////////////////////////`  

` match(Do);`  
` a = text +1;`  

` //statement();`  

` match('{');`  
` // statements`  
` while (token != '}') {`  
` statement();`  
` }`  
` match('}');`  

` match(While);`  

` match('(');`  

` expression(Assign);`  
` match(')');`  

` match(';');`  

` *++text = JNZ;`  
` *++text = (int)a;`  

<br />

` }`

**Comment by icaiyu** [comment]
有个疑问啊，在function_body函数的最后已经有*++text = LEV; 但是在statement函数的Return分支的最后为什么还要有*++text = LEV;，不会造成重复吗？
